test/org/freedesktop/gstreamer/PropertyTypeTest.java

```diff
@@ -0,0 +1,172 @@
+        assertEquals(0, audiotestsrc.get("blocksize"));
+
+        audiotestsrc.set("blocksize", 42);
+        assertEquals(42, audiotestsrc.get("blocksize"));
+
+        int maxUnsignedInt = Integer.parseUnsignedInt("4294967295");
+        audiotestsrc.set("blocksize", maxUnsignedInt);
+        assertEquals(maxUnsignedInt, audiotestsrc.get("blocksize"));
+    }
+
+    @Test
+    public void setLong() {
+        audiotestsrc.set("timestamp-offset", 0L);
+        assertEquals(0L, audiotestsrc.get("timestamp-offset"));
+
+        audiotestsrc.set("timestamp-offset", 42L);
+        assertEquals(42L, audiotestsrc.get("timestamp-offset"));
+
+        audiotestsrc.set("timestamp-offset", -42L);
+        assertEquals(-42L, audiotestsrc.get("timestamp-offset"));
+
+        audiotestsrc.set("timestamp-offset", 9223372036854775807L);
+        assertEquals(9223372036854775807L, audiotestsrc.get("timestamp-offset"));
+
+        audiotestsrc.set("timestamp-offset", -9223372036854775808L);
+        assertEquals(-9223372036854775808L, audiotestsrc.get("timestamp-offset"));
+    }
+
+    @Test
+    public void setULong() {
+        filesink.set("max-bitrate", 0L);
+        assertEquals(0L, filesink.get("max-bitrate"));
+
+        long maxUnsignedLong = Long.parseUnsignedLong("18446744073709551615");
+        filesink.set("max-bitrate", maxUnsignedLong);
+        assertEquals(maxUnsignedLong, filesink.get("max-bitrate"));
+        
+        filesink.set("max-bitrate", 42L);
+        assertEquals(42L, filesink.get("max-bitrate"));
+
+        filesink.setAsString("max-bitrate", "18446744073709551615");
+        assertEquals(maxUnsignedLong, filesink.get("max-bitrate"));
+    }
+
+    @Test
+    public void setString() {
+        filesink.set("location", "");
+        assertEquals("", filesink.get("location"));
+
+        filesink.set("location", null);
+        assertNull(filesink.get("location"));
+
+        filesink.set("location", "foobar");
+        assertEquals("foobar", filesink.get("location"));
+    }
+
+    @Test
+    public void setEnum() {
+        audiotestsrc.set("wave", 8);
+        assertEquals(8, audiotestsrc.get("wave"));
+
+        audiotestsrc.setAsString("wave", "Silence");
+        assertEquals(4, audiotestsrc.get("wave"));
+        assertEquals("Silence", audiotestsrc.getAsString("wave"));
+        
+        audiotestsrc.setAsString("wave", "square");
+        assertEquals(1, audiotestsrc.get("wave"));
+        assertEquals("Square", audiotestsrc.getAsString("wave"));
+        
+        audiotestsrc.setAsString("wave", "red-noise");
+        assertEquals(10, audiotestsrc.get("wave"));
+        String redNoise = audiotestsrc.getAsString("wave");
+        assertEquals("Red (brownian) noise", redNoise);
+        audiotestsrc.setAsString("wave", redNoise);
+        assertEquals(10, audiotestsrc.get("wave"));
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void setEnumInvalid() {
```

Comment:
The Test is more about `settingEnumToStringValueIsInvalid`; `setEnumInvalid` sounds like setting an Enum to an invalidValue – da Test that would certainly also be good to have here.

Reply:
Will consider test name change. Not sure what second point means - in what way setting an enum to an invalid value? The property? Is that possible?

Reply:
The second paragraph requests a Test that documents the (correct) behavior when calling setAsString and set for an enum with a not valid value, ie setting the `wave`-property on an `audiotestsrc` to `yoda`.

Reply:
But how is that different to what's already there?

Reply:
It would appear I'm an idiot! :laughing: That was meant to be a test of `setAsString`. Fixed.